pyteal/ast/subroutine_test.py

```diff
@@ -88,6 +88,13 @@ def fnWithExprAnnotations(a: Expr, b: Expr) -> Expr:
     def fnWithSVAnnotations(a: ScratchVar, b: ScratchVar):
         return Return()
 
+    def fnWithABIAnnotations(
+        a: abi.Byte,
+        b: abi.StaticArray[abi.Uint32, Literal[10]],
+        c: abi.DynamicArray[abi.Bool],
+    ):
```

Comment:
```python
    def fnWithABIAnnotations(
        a: abi.Byte,
        b: abi.StaticArray[abi.Uint32, Literal[10]],
        c: abi.DynamicArray[abi.Bool],
    ):
```

We have a bit of a race condition between this PR and #260. Here are the issues:
* It seems like Blackbox testing is important for vetting the semantic effects of this work
* But PR #260 supports a different format for specifying the input type, rather than using generics/templats as is done here
* It seems to me that this approach is better than that of #260

I propose that this PR go in first, and that #260 adopt the approach taken here. But that in short order, after #260 is merged a PR whose main goal is to semantically test the ABI router along with other types of subroutines (to ensure nothing broke) be created in short order.

Reply:
good point. I guess what we can do now is to set a reminder here indicating the situation (like a TODO in code snippet), in case we would forget about sync both implementations.